tests/distribution/python_version_change_tests/common.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2014-2020 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+import time
+import json
+
+from tests.utils.compat import Path
+from tests.utils.agent_runner import AgentRunner, PACKAGE_INSTALL
+
+
+def _get_python_major_version(runner):
+    status = json.loads(runner.status_json())
+
+    version_string = status["python_version"]
+
+    version = int(version_string[0])
+
+    return version
+
+
+def common_test_python3(install_package_fn):
+    """
+    Test package installation on machine with python3
+    :param install_package_fn: callable that installs package with appropriate type to the current machine OS.
+    """
+
+    # remove python and python2 to make installer see only python3
+    try:
+        # some distros do not have 'python' command.
+        os.remove(str(Path("/", "usr", "bin", "python")))
+    except IOError:
+        pass
+    os.remove(str(Path("/", "usr", "bin", "python2")))
+
+    install_package_fn()
+
+    runner = AgentRunner(PACKAGE_INSTALL)
+    runner.start()
+    time.sleep(1)
```

Comment:
I guess we should just keep an eye on this sleep if we encounter any races because the agent hasn't fully start up yet in the future.